data/creaturescripts/scripts/droploot.lua

```diff
@@ -21,15 +22,27 @@ function onDeath(player, corpse, killer, mostDamage, unjustified, mostDamage_unj
 			player:removeItem(ITEM_AMULETOFLOSS, 1, -1, false)
 		end
 	else
-		for i = CONST_SLOT_HEAD, CONST_SLOT_AMMO do
-			local item = player:getSlotItem(i)
-			if item then
-				if table.contains({SKULL_RED, SKULL_BLACK}, player:getSkull()) or math.random(item:isContainer() and 100 or 1000) <= player:getLossPercent() then
+		if isRedOrBlack then 
+			for i = CONST_SLOT_HEAD, CONST_SLOT_AMMO do
+				local item = player:getSlotItem(i)
+				if item then
 					if not item:moveTo(corpse) then
 						item:remove()
 					end
 				end
 			end
+		else 
+			for i = CONST_SLOT_HEAD, CONST_SLOT_AMMO do
+				local item = player:getSlotItem(i)
+				local lossPercent = player:getLossPercent()
+				if item then
+					if math.random(item:isContainer() and 100 or 1000) <= lossPercent then
```

Comment:
This could be:
```suggestion
					if isRedOrBlack or math.random(item:isContainer() and 100 or 1000) <= lossPercent then
```
and then the previous `if isRedOrBlack then...` is unnecessary as it's effectively the same code duplicated.

By the way, this function doesn't seem entirely correct, when lossPercent is 0, the random value could still be 0 and player would lose the item despite supposedly having 0% chance. There's <1% chance of that but still.

Reply:
Done.

I guess I exagerated on not checking the same thing multiple times.

Also fixed the 0% lossPercent chance.